src/pages/home/report/ReportActionsView.js

```diff
@@ -338,6 +342,11 @@ class ReportActionsView extends React.Component {
     }
 
     render() {
+        // If the `isChatReportEmptyWhileOffline` prop is true then let's render ReportActionItemCreated component
+        if (this.props.isChatReportEmptyWhileOffline) {
+            return <ReportActionItemCreated reportID={this.props.report.reportID} />;
+        }
```

Comment:
This works, but I think is going to be a bit weird UX. As soon as someone adds an optimistic action while offline this will disappear since `reportActions.length > 0`.

Reply:
But do we have any other way out? Because it is bound to happen, right? Can we show some additional message to let user know we don't have data cached?

Reply:
Yeah, very tricky. So to summarize:

- We can tell a report has not loaded yet if it has `0` `reportActions.length`
- If we are offline then we can show this new special "created" action 
- As soon as they add a comment it will look like the report has loaded

So the question becomes...

**How can we tell that we need to keep showing the "created" action if we are depending on `reportActions.length` and adding optimistic actions will increase this length?**

First solution that comes to mind is something like...

```js
const hasAtLeastOneNonPendingAction = reportActions.some(action => !action.pendingAction);
```

Which would be the same as "empty or empty with only pending actions". What do you think?

Reply:
I am not sure if I am thinking in the right direction or not but the problem with `some` will be that even if a chat report is loaded and we added an optimistic action to the report the above will evaluate to `true` right? So it all boils down to all the report actions must be in a pending state or the list should be empty inorder for us to show the `created` action, no?

I also went through the other [issue](https://github.com/Expensify/App/issues/13074) which is very similar to this one and I really liked the idea of showing `[broken cloud] Let's sync up when you get back online` rather than a `created` action because both seem to be aligned with our optimistic approach and the latter one defines the situation better here.

Reply:
>the problem with some will be that even if a chat report is loaded and we added an optimistic action to the report the above will evaluate to true right

My assumption is that if the report has truly loaded it will have a "created" action that is not pending.

>all report actions must be in a pending state or the list should be empty in order for us to show the created action, no

Yep, I think that makes sense.

Reply:
Yeah, I agree we can show the skeleton UI without animation rather than a created action.